### ISSUE TITLE
BUG: ArrowDtype Series: Broadcasting Error with Single-Value Arrays in Arithmetic Operations

### ISSUE THREAD

**Issue by hombit** [pandas-dev/pandas#60073]
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [X] I have confirmed this bug exists on the [main branch](https://pandas.pydata.org/docs/dev/getting_started/install.html#installing-the-development-version-of-pandas) of pandas.

### Reproducible Example

```python
import numpy as np
import pandas as pd
import pyarrow as pa

series = pd.Series([1.0, 2.0, 3.0], dtype=pd.ArrowDtype(pa.float64()))
series / np.array([2.0])
# ArrowInvalid: Array arguments must all be the same length
```

### Issue Description

When performing arithmetic operations between a `Series` of `ArrowDtype` and a single-value array, the array is not broadcasted to match the series shape. Instead, it raises an `ArrowInvalid error`. This behavior is inconsistent, as the same operation works correctly with both default and numpy-nullable dtypes.

A solution could be casting a single-element array to a `pa.Scalar` in `ArrowExtensionArray._evaluate_op_method`:
https://github.com/pandas-dev/pandas/blob/2a10e04a099d5f1633abcdfbb2dd9fdf09142f8d/pandas/core/arrays/arrow/array.py#L755

### Expected Behavior

The code doesn't fail, and behavior to be consistent over different dtype back-ends

### Installed Versions

Pandas v2.2.3:

<details>

INSTALLED VERSIONS
------------------
commit                : 0691c5cf90477d3503834d983f69350f250a6ff7
python                : 3.12.7
python-bits           : 64
OS                    : Darwin
OS-release            : 24.0.0
Version               : Darwin Kernel Version 24.0.0: Tue Sep 24 23:37:36 PDT 2024; root:xnu-11215.1.12~1/RELEASE_ARM64_T6020
machine               : arm64
processor             : arm
byteorder             : little
LC_ALL                : en_US.UTF-8
LANG                  : en_US.UTF-8
LOCALE                : en_US.UTF-8
pandas                : 2.2.3
numpy                 : 1.26.4
pytz                  : 2024.1
dateutil              : 2.9.0.post0
pip                   : 24.2
Cython                : None
sphinx                : 7.4.7
IPython               : 8.26.0
adbc-driver-postgresql: None
adbc-driver-sqlite    : None
bs4                   : 4.12.3
blosc                 : None
bottleneck            : None
dataframe-api-compat  : None
fastparquet           : None
fsspec                : 2024.6.1
html5lib              : None
hypothesis            : None
gcsfs                 : None
jinja2                : 3.1.4
lxml.etree            : None
matplotlib            : 3.8.4
numba                 : 0.60.0
numexpr               : None
odfpy                 : None
openpyxl              : None
pandas_gbq            : None
psycopg2              : None
pymysql               : None
pyarrow               : 16.1.0
pyreadstat            : None
pytest                : 8.3.2
python-calamine       : None
pyxlsb                : None
s3fs                  : 2024.6.1
scipy                 : 1.14.0
sqlalchemy            : None
tables                : None
tabulate              : 0.9.0
xarray                : None
xlrd                  : None
xlsxwriter            : None
zstandard             : None
tzdata                : 2024.1
qtpy                  : None
pyqt5                 : None

</details>

Nightly Pandas:

<details>
INSTALLED VERSIONS
------------------
commit                : 2a10e04a099d5f1633abcdfbb2dd9fdf09142f8d
python                : 3.12.7
python-bits           : 64
OS                    : Darwin
OS-release            : 24.0.0
Version               : Darwin Kernel Version 24.0.0: Tue Sep 24 23:37:36 PDT 2024; root:xnu-11215.1.12~1/RELEASE_ARM64_T6020
machine               : arm64
processor             : arm
byteorder             : little
LC_ALL                : en_US.UTF-8
LANG                  : en_US.UTF-8
LOCALE                : en_US.UTF-8

pandas                : 3.0.0.dev0+1579.g2a10e04a09
numpy                 : 2.2.0.dev0+git20241015.961b70f
dateutil              : 2.9.0.post0
pip                   : 24.2
Cython                : None
sphinx                : None
IPython               : 8.28.0
adbc-driver-postgresql: None
adbc-driver-sqlite    : None
bs4                   : None
blosc                 : None
bottleneck            : None
fastparquet           : None
fsspec                : None
html5lib              : None
hypothesis            : None
gcsfs                 : None
jinja2                : None
lxml.etree            : None
matplotlib            : None
numba                 : None
numexpr               : None
odfpy                 : None
openpyxl              : None
psycopg2              : None
pymysql               : None
pyarrow               : 17.0.0
pyreadstat            : None
pytest                : None
python-calamine       : None
pytz                  : None
pyxlsb                : None
s3fs                  : None
scipy                 : None
sqlalchemy            : None
tables                : None
tabulate              : None
xarray                : None
xlrd                  : None
xlsxwriter            : None
zstandard             : None
tzdata                : 2024.2
qtpy                  : None
pyqt5                 : None
</details>

**Comment by rhshadrach** [comment]
Agreed this is inconsistent, and the fix seems reasonable. cc @WillAyd @jorisvandenbossche

**Comment by WillAyd** [comment]
I think this is fine to support, although its probably better changing upstream in Arrow than implementing a fix here directly

**Comment by hombit** [comment]
@WillAyd it doesn't look like Arrow has an idea of array broadcasting, do you think I should open a feature issue there? Do you think it is likely to be implemented?

**Comment by WillAyd** [comment]
Probably worth asking there first. If it doesn't make sense can come back here for implementation

**Comment by hombit** [comment]
@WillAyd I asked, and it is about to automatically close after a year of inactivity. =(
https://github.com/apache/arrow/issues/44544

**Comment by thisisnic** [comment]
@hombit I've unmarked it as stale on Arrow as it seems worth us looking into, will ping a relevant maintainer to get their thoughts on it.
